### ISSUE TITLE
psr

### ISSUE THREAD

**Issue by daveismyname** [simple-mvc-framework/framework#120]
Hi guys,

Recently I've been thinking about following a coding standard, what are peoples thoughts on implementing PSR-2? 

https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md

http://www.phptherightway.com/#code_style_guide

Also should we adopt psr-4 autoloading?

https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-4-autoloader.md

**Comment by SecretD** [comment]
It would be nice to use the PSR-2 coding standard. It will avoid the case-sensitivity issues with Linux and you can check the PSR-2 standard with [PHP-CS](https://github.com/squizlabs/PHP_CodeSniffer) we should also adopt the PSR-4 autoloading. And what about testing the code?

**Comment by Volter9** [comment]
Isn't framework following PSR-4 autoloading already?:
```json
{
    ...
    "autoload": {
        "psr-4": {"": "app/"}
    }
    ...
}
```
It would be also nice if framework was in separate folder from app (with its own namespace).

**Comment by daveismyname** [comment]
@Volter9 ah yes I thought it was using PSR-0 for some odd reason. As for separating the framework it's something to think about, curious what you you have in the separate folders?
@SecretD Thats my thoughts too it will also make it more compatible with other components/third party libraries.

For testing I'm note sure yet I haven't used PHPUnit yet it's on my list of things I should use.

I would release this as a new release that way to keep the current version available for anyone that does not want to use PSR.

**Comment by SecretD** [comment]
@daveismyname Most of the times I use codeception for my projects written in SMVC it is a nice way to test your applications and it supports unit testing.

**Comment by stnc** [comment]
core / view.php 
add to exract ($data);
array to varible .....

**Comment by daveismyname** [comment]
Hi guys been looking over PSR more closely I'm looking to release a new release over the coming weeks, it will be tagged as 2.1.1 

As the convention for names is camel case a bunch of methods will need to be renamed to comply to that end new docs will be written details the changes as well as the usual doc info.

I can quite decide if views files should be camel cased too I'm thinking they should be to be consistent with the class files.

I'm rather new to PSR myself so any thought or preferences let me know.

**Comment by daveismyname** [comment]
2.2 has now been released main repo and docs updated, the framework now fully supports PSR 1, 2 and 4

**Comment by Volter9** [comment]
:+1:

**Comment by ageddesi** [comment]
Double thumbs up

**Comment by autotim** [comment]
great work :star2::star2::star2::star2::star2: :+1:

**Comment by jimgwhit** [comment]
Dave, will there be docs included in the framework like v2.1 had? And if I want to contribute or add to online docs, where do I send my addition for your review and addition?

**Comment by daveismyname** [comment]
Yes I'll be finishing the docs over the next few days for now you can email me with and amendments or additions Dave@simplemvcframework.com

Sent from my HTC

----- Reply message -----
